### ISSUE TITLE
Не работает авторизация

### ISSUE THREAD

**Issue by amidon** [LTD-Beget/sprutio#83]
Установил sprut.io. Все было без ошибок. Но авторизация не проходит и ошибок не выдает просто после нажатия на вход идет Loading после обратно кнопка войти? Что может быть не так?

**Comment by shadowlmd** [comment]
Логи надо смотреть. Скорее всего по какой-то причине у контейнера rpc нет доступа к системным /etc/passwd и /etc/shadow.

**Comment by yazux** [comment]
Та же самая проблема. Авторизация внезапно перестала работать, хотя ранее было всё отлично.

**Comment by shadowlmd** [comment]
Зря вы так логи в открытую выкладываете, там же все ваши пароли в открытом виде. По проблеме, видно, что авторизация не проходит, как будто пароль неправильный. Скиньте вывод `docker inspect sprutio_rpc_1`.

**Comment by yazux** [comment]
Пароль то я поменяю, понятное дело) Сейчас срочно нужен ваш менеджер.
Лог прилагаю.
[log.txt](https://github.com/LTD-Beget/sprutio/files/1788861/log.txt)

**Comment by shadowlmd** [comment]
Образ выглядит нормально. Перезапуск не помогает?

**Comment by yazux** [comment]
Нет, север перезагружал, sprut перезапускал, переустанавливал. Не помогло.

**Comment by yazux** [comment]
Лог после перезагрузки менеджера
[log.txt](https://github.com/LTD-Beget/sprutio/files/1788876/log.txt)

**Comment by shadowlmd** [comment]
Тогда запустите `docker exec -t -i sprutio_rpc_1 bash` и там уже вот эти команды:
```
ls -l /etc/passwd /etc/shadow
cat /etc/passwd
cat /etc/shadow
```
В /etc/shadow содержатся хеши паролей всех пользователей системы, так что если там есть живые пользователи, лучше сюда не выкладывать вывод последней команды.

**Comment by yazux** [comment]
Проделал, хеши заменил на рандомные символы.
[log.txt](https://github.com/LTD-Beget/sprutio/files/1788953/log.txt)

**Comment by shadowlmd** [comment]
А давно эти файлы стали симлинками? Не в этот ли момент всё сломалось?
```
lrwxrwxrwx 1 root root 15 Mar  7 11:04 /etc/passwd -> /mnt/etc/passwd
lrwxrwxrwx 1 root root 15 Mar  7 11:04 /etc/shadow -> /mnt/etc/shadow
```
И можно тогда ещё `ls -l /mnt/etc/{passwd,shadow}`, не обязательно файлом, можно прямо сюда.

**Comment by yazux** [comment]
Сервер только вчера был сконфигурирован, эти файлы ни кто не трогал. Если они и стали смлинками, то только в результате установки и запуска какого-либо пакета.
Из последнего устанавливалось только: redis и unzip

root@0d5630768b24:/rpc# ls -l /mnt/etc/{passwd,shadow}
-rwxrwxr-x 1 root root   1516 Mar  6 09:25 /mnt/etc/passwd
-rwxrwxr-x 1 root shadow  938 Mar  6 09:25 /mnt/etc/shadow

**Comment by shadowlmd** [comment]
Жесть какая. Пермишны на /mnt/etc/shadow должны быть 640, а не 664. Исправить надо в любом случае, хотя не факт, что поможет. Если не поможет, нужно убрать симлинки и положить вместо них файлы.

**Comment by yazux** [comment]
В папке mnt пусто
root@xyazu:~# cd /mnt
root@xyazu:/mnt# ls
root@xyazu:/mnt# 

На этих файлах в папке /etc права 775
Но тут по 2 файла:
passwd
passwd-
shadow
shadow-

**Comment by shadowlmd** [comment]
Это я ступил, в докере и должны быть симлинки. Нужно просто сделать нормальные права на /etc/shadow. Должно быть вот так:
```
# ls -la /mnt/etc/{passwd,shadow}
-rw-r--r-- 1 root root   279 Mar  7 12:23 /mnt/etc/passwd
-rw-r----- 1 root shadow 449 Mar  7 12:23 /mnt/etc/shadow
```

**Comment by yazux** [comment]
Поменял права, ситуация не изменилась

**Comment by yazux** [comment]
root@xyazu:~# docker exec -t -i sprutio_rpc_1 bash
root@629b82758e47:/rpc# ls -la /mnt/etc/{passwd,shadow}
-rw-r----- 1 root root   1516 Mar  6 09:25 /mnt/etc/passwd
-rw-r----- 1 root shadow  938 Mar  6 09:25 /mnt/etc/shadow
root@629b82758e47:/rpc#

**Comment by shadowlmd** [comment]
У passwd должно быть 644. :) Но по идее на авторизацию Spritio это не должно было повлиять. Глупый вопрос, но пароль точно правильный?

**Comment by yazux** [comment]
Да, пароль верный. Проверено было уже раз 50 =)

**Comment by yazux** [comment]
Если есть возможность, можете написать в телеграм: @xyazu, дам ссылку на сервер

**Comment by shadowlmd** [comment]
У меня пока что нет времени настолько глубоко вникать в проблему (я не разработчик, если что, обычный юзер). Я знаю что для авторизации Sprutio нужны только /etc/passwd и /etc/shadow, и по идее если система может авторизовать по логину/паролю, то и Sprutio может. Мой допиленный вариант вообще не имеет доступа к файловой системе сервера, я ему подсовываю практически пустую директорию, в которой есть etc с файлами passwd, shadow и group, в которых есть только записи для юзеров, которым можно логиниться в Sprutio. Как вариант, можно создать обычного пользователя с каким-нибудь простым паролем, и попробовать залогиниться им. Если не получится, скинуть сюда записи из passwd и shadow для этого юзера, и пароль, и я попробую проверить этого юзера у себя.

**Comment by yazux** [comment]
В общем, всё решилось весьма печальным способом, но единственно возможным в моём случае:

Потребовалось установить панель Vesta, а она требует чистый сервер для установки. Пришлось  сбросить его, накатить панель, а потом sprut. Всё завелось нормально.

**Comment by bezumkin** [comment]
Лично мне в подобном случае помогло `docker-compose -p sprutio up -d`
